File: test/language/templates/cpp/InstantiateImportedTest.cpp
#include "gtest/gtest.h"

#include "templates/instantiate_imported/InstantiateImported.h"

namespace templates
{
namespace instantiate_imported
{

TEST(InstantiateImportedTest, readWrite)
{
    InstantiateImported instantiateImported;
    instantiateImported.setTest32(pkg::U32{13});
    instantiateImported.setTestStr(Test_string{"test"});

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiateImported.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiateImported readInstantiateImported(reader);

    ASSERT_TRUE(instantiateImported == readInstantiateImported);
}

} // namespace instantiate_imported
} // namespace templates

File: test/language/templates/python/InstantiateImportedTest.py
import zserio

import Templates

class InstantiateImportedTest(Templates.TestCase):
    def testReadWrite(self):
        instantiateImported = self.api.InstantiateImported(self.api.pkg.U32(13), self.api.Test_string("test"))

        writer = zserio.BitStreamWriter()
        instantiateImported.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readInstantiateImported = self.api.InstantiateImported()
        readInstantiateImported.read(reader)
        self.assertEqual(instantiateImported, readInstantiateImported)
